gui/window.cpp

```diff
@@ -931,6 +936,39 @@ void Window::sshConsole()
 #endif
 }
 
+#if __APPLE__
+void Window::hyperkitPermission()
+{
+    QString command = "sudo chown root:wheel ~/.minikube/bin/docker-machine-driver-hyperkit && "
+                      "sudo chmod u+s ~/.minikube/bin/docker-machine-driver-hyperkit && exit";
+    QStringList arguments = { "-e", "tell app \"Terminal\"",
+                              "-e", "set w to do script \"" + command + "\"",
+                              "-e", "activate",
+                              "-e", "repeat",
+                              "-e", "delay 0.1",
+                              "-e", "if not busy of w then exit repeat",
+                              "-e", "end repeat",
+                              "-e", "end tell" };
+    QProcess *process = new QProcess(this);
+    process->start("/usr/bin/osascript", arguments);
+    process->waitForFinished(-1);
+}
+
+void Window::showHyperKitMessage()
+{
+    QMessageBox dialog;
+    dialog.setWindowTitle(tr("HyperKit Permissions Required"));
+    dialog.setWindowIcon(*trayIconIcon);
+    dialog.setModal(true);
+    dialog.setText(tr(
```

Comment:
"The HyperKit driver requires a one-time sudo permission, If you like to proceed, Press OK and then enter your password in the terminal the start will resume after."

Reply:
there should be a cancel button to not continue.

Reply:
there should be a cancel button to not continue.

Reply:
Message updated and cancel button added.